src/components/Drawer/components/Drawer.scss

```diff
@@ -0,0 +1,211 @@
+    &[data-floating-ui-status='close'] &__item_direction_right {
+        transform: translate(100%, 0);
+    }
+
+    /* top */
+    &[data-floating-ui-status='initial'] &__item_direction_top,
+    &[data-floating-ui-status='close'] &__item_direction_top {
+        transform: translate(0, -100%);
+    }
+
+    /* bottom */
+    &[data-floating-ui-status='initial'] &__item_direction_bottom,
+    &[data-floating-ui-status='close'] &__item_direction_bottom {
+        transform: translate(0, 100%);
+    }
+
+    &__veil {
+        --_--veil-background-color: var(--g-color-sfx-veil);
+
+        position: absolute;
+        z-index: var(--g-drawer-veil-z-index, 100);
+        inset: 0;
+
+        background-color: var(--g-drawer-veil-background-color, var(--_--veil-background-color));
+
+        pointer-events: initial;
+
+        &_hidden {
+            background-color: transparent;
+        }
+    }
+
+    &:not(&_skip-animation)[data-floating-ui-status='open'] &__veil,
+    &:not(&_skip-animation)[data-floating-ui-status='close'] &__veil {
+        transition: opacity 300ms;
+    }
+
+    &[data-floating-ui-status='initial'] &__veil,
+    &[data-floating-ui-status='close'] &__veil {
+        opacity: 0;
+    }
+
+    &[data-floating-ui-status='unmounted'] &__veil {
+        visibility: hidden;
+    }
+
+    &[data-floating-ui-status='open'] &__veil {
+        opacity: 1;
+    }
+
+    &__resizer-handle {
+        width: 2px;
+        height: 28px;
+
+        background: var(--g-drawer-item-resizer-handle-color, var(--_--resizer-handle-color));
+        border-radius: 2px;
+
+        &_direction_top,
+        &_direction_bottom {
+            width: 28px;
+            height: 2px;
+        }
+    }
+
+    &__resizer {
+        position: absolute;
+        inset-block-start: 0;
+        z-index: var(--g-drawer-item-resizer-z-index, var(--_--resizer-z-index));
+
+        display: flex;
+        flex-direction: column;
+        justify-content: center;
+        align-items: center;
+
+        width: var(--g-drawer-item-resizer-width, var(--_--resizer-width));
+        height: 100%;
+
+        cursor: col-resize;
+
+        background: var(--g-drawer-item-resizer-color, var(--_--resizer-color));
+
+        &_direction_right {
+            inset-inline-start: 0;
+            inset-block-start: var(--g-drawer-item-top-offset, 0);
+        }
+
+        &_direction_left {
+            inset-inline-end: 0;
+            inset-block-start: var(--g-drawer-item-top-offset, 0);
+        }
+
+        &_direction_top {
+            inset-block: auto 0;
+            inset-inline-start: 0;
+        }
+
+        &_direction_bottom {
+            inset-inline-start: 0;
+            inset-block-start: 0;
+        }
+
+        &_direction_top,
+        &_direction_bottom {
+            cursor: row-resize;
+            width: 100%;
+            height: var(--g-drawer-item-resizer-width, var(--_--resizer-width));
```

Comment:
If name used for both width and height, then it should be called like `*-size`, otherwise it's confusing